src/views/TimezoneModal.vue

```diff
@@ -0,0 +1,171 @@
+<template>
+  <ion-header>
+    <ion-toolbar>
+      <ion-buttons slot="start">
+        <ion-button @click="closeModal"> 
+          <ion-icon :icon="close" />
+        </ion-button>
+      </ion-buttons>
+      <ion-title>{{ $t("Select time zone") }}</ion-title>
+    </ion-toolbar>
+    <ion-toolbar>
+      <ion-searchbar @ionFocus="selectSearchBarText($event)" :placeholder="$t('Search time zones')"  v-model="queryString" @ionInput="findTimeZone()"></ion-searchbar>
+    </ion-toolbar>
+  </ion-header>
+
+  <ion-content class="ion-padding">
+    <!-- Empty state -->
+    <div class="empty-state" v-if="filteredTimeZones.length === 0">
+      <p>{{ $t("No time zone found")}}</p>
+    </div>
+
+    <!-- Timezones -->
+    <div v-else>
+      <ion-list>
+        <ion-radio-group value="rd" v-model="timeZoneId">
+          <ion-item  v-bind:key="timeZone.id" v-for="timeZone in filteredTimeZones">
+            <ion-label>{{ timeZone.label }} ({{ timeZone.id }})</ion-label>
+            <ion-radio :value="timeZone.id" slot="start"></ion-radio>
+          </ion-item>
+        </ion-radio-group>
+      </ion-list>
+    </div>
+    
+    <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+      <ion-fab-button :disabled="!timeZoneId" @click="saveAlert">
+        <ion-icon :icon="save" />
+      </ion-fab-button>
+    </ion-fab>
+  </ion-content>
+</template>
+
+<script lang="ts">
+import { 
+  IonButtons,
+  IonButton,
+  IonContent,
+  IonFab,
+  IonFabButton,
+  IonHeader,
+  IonItem,
+  IonIcon,
+  IonLabel,
+  IonRadioGroup,
+  IonRadio,
+  IonList,
+  IonSearchbar,
+  IonTitle,
+  IonToolbar,
+  modalController,
+  alertController } from "@ionic/vue";
+import { defineComponent } from "vue";
+import { close, save } from "ionicons/icons";
+import { useStore } from "@/store";
+import { UserService } from "@/services/UserService";
+import { hasError } from '@/utils'
+
+export default defineComponent({
+  name: "TimeZoneModal",
+  data() {
+    return {
+      queryString: '',
+      filteredTimeZones: [],
+      timeZones: [],
+      timeZoneId: ''
+    }
+  },
+  computed: {
```

Comment:
Removed the computed block as its not needed.

Reply:
Removed.